src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAwaitForeachTests.cs

```diff
@@ -8023,5 +8023,28 @@ public static class Extensions
             comp.VerifyDiagnostics();
             CompileAndVerify(comp, expectedOutput: "123123");
         }
+
+        [Fact]
+        [WorkItem(53426, "https://github.com/dotnet/roslyn/issues/53426")]
+        public void TestAwaitAsyncEnumerableNoGetAwaiterErrorMessage()
+        {
+            var source = @"
+using System.Collections.Generic;
+using System.Threading.Tasks;
+public class C 
+{
+    public async Task M(IAsyncEnumerable<int> e) 
+    {
+        await e; 
+    }
+}";
+
+            var expected = Diagnostic(ErrorCode.ERR_NoAwaitOnAsyncEnumerable, "await e")
+                .WithArguments("System.Collections.Generic.IAsyncEnumerable<int>", "GetAwaiter")
+                .WithLocation(8, 9);
+
+            CreateCompilationWithTasksExtensions(new[] { source, s_IAsyncEnumerable }, parseOptions: TestOptions.Regular8)
+                .VerifyDiagnostics(expected);
```

Comment:
We tend to use a different format for `VerifyDiagnostics`. See `TestWithCSharp7_3` for example. Something like:
```
            comp.VerifyDiagnostics(
                // (7,9): error CS8652: The feature 'async streams' is not available in C# 7.3. Please use language version 8.0 or greater.
                //         await foreach (int i in new C())
                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion7_3, "await").WithArguments("async streams", "8.0").WithLocation(7, 9)
                );
```

To achieve that, you can just let `VerifyDiagnostics` fail and copy&paste from the test failure output (it should contain the "expected" blurb).